.github/workflows/release.yml

```diff
@@ -3,16 +3,20 @@ name: Release public Lambda layer
 on:
   workflow_dispatch:
     inputs:
-      source_layer_arn:
-        description: 'Clone layer from ARN: (find layer ARN from soaking workflow log)'
+      layer_arn_keyword:
+        description: 'Clone layer from keyword ARN by substituting architecture parameter (ie: arn:aws:lambda:us-east-1:012345678912:layer:aws-otel-nodejs-wrapper-<ARCHITECTURE>-d8b20954c3284abcd08f5fb8178986a89cff7160:1)'
         required: true
-      layer_name:
-        description: 'Publish to layer name (ie: aws-otel-python38-ver-1-1-0)'
+      layer_name_keyword:
+        description: 'Publish layer to keyword name by substituting architecture parameter (ie: aws-otel-python38-<ARCHITECTURE>-ver-1-7-1)'
         required: true
       aws_region:
         description: 'Deploy to aws region'
         required: true
         default: "[ \"us-east-1\", \"us-east-2\", \"us-west-1\", \"us-west-2\", \"ap-south-1\", \"ap-northeast-2\", \"ap-southeast-1\", \"ap-southeast-2\", \"ap-northeast-1\", \"ca-central-1\", \"eu-central-1\", \"eu-west-1\", \"eu-west-2\", \"eu-west-3\", \"eu-north-1\", \"sa-east-1\" ]"
+      architecture:
+        description: 'Architecture-compatible distributions of the layer to release in each region'
+        required: true
+        default: "[ \"amd64\", \"arm64\" ]"
```

Comment:
so, we are bulding layer for both arch by default? or should we take as an input on which layer to build? Taking an input would be specifically helpful if for some reason we need to update/fix only one of the arch layer.

Reply:
> so, we are bulding layer for both arch by default? or should we take as an input on which layer to build?

Yes and yes to both questions! 🙂 By default we will release both architectures.

I thought this was a good idea since the Soak Tests will ALWAYS Soak Test both architectures. i.e. you cannot **soak test** just _one_ architecture, and if you're making code changes we probably want to always test both architectures anyways.

https://github.com/aws-observability/aws-otel-lambda/blob/3673ac5b0c02495b25db392f880e52e618e2f034/.github/workflows/soaking.yml#L32

However, you might want to only **release** _one_ architecture. So that's why we _allow_ the input to be changed, but by default we leave it to be all architectures that we _expect_ the Soak Test "Layer ARN Keyword" is valid for.

I think the operator should be extra cautious if they are trying to modify the release "architecture" input, because the only reason they would is if the Soak Test Layer ARN Keyword only worked for ONE architecture and FAILED for the other and at the point we should be really certain that we want to release that kind of layer.

Reply:
right. Thanks for an explanation! Do you want to add the last part as comment in the workflow? What should operator be extra cautious about?

Reply:
Sure I think that would be helpful! I added the comment 🙂